File: app/src/main/java/com/example/protect/model/DataModels.kt
package com.example.protect.model

import androidx.compose.ui.graphics.vector.ImageVector

enum class UserRole(
    val title: String,
    val shortName: String,
    val subtitle: String,
    val iconEmoji: String,
    val businessModel: String
) {
    B2C_RETAIL(
        title = "Pengguna Retail (B2C)",
        shortName = "Retail B2C",
        subtitle = "Rumah, Ruko & Apartemen • Pemesanan Cepat & E-Garansi",
        iconEmoji = "🏠",
        businessModel = "Model Transaksional: Bayar per-kunjungan treatment atau langganan pencegahan berkala."
    ),
    B2B_ENTERPRISE(
        title = "Perusahaan Korporat (B2B)",
        shortName = "Enterprise B2B",
        subtitle = "Pabrik, Gudang & F&B • CCTV ProViewAI, HACCP & ESG",
        iconEmoji = "🏢",
        businessModel = "Model SaaS + Hardware: Biaya langganan software AI monitoring CCTV & audit berkala."
    ),
    FIELD_TECHNICIAN(
        title = "Eksekutor Lapangan (Teknisi)",
        shortName = "Field Tech",
        subtitle = "Petugas Pest Control • GPS Check-In, SOP & E-Sign",
        iconEmoji = "👷‍♂️",
        businessModel = "Operasional Lapangan: Pelaksanaan SOP keselamatan, chemical logging, dan berita acara digital."
    )
}

data class ServiceItem(
    val id: String,
    val title: String,
    val subtitle: String,
    val price: Long,
    val description: String = ""
)

data class IncidentItem(
    val id: String,
    val code: String,
    val title: String,
    val timestamp: String,
    val location: String,
    val severity: SeverityLevel,
    val status: IncidentStatus,
    val confidence: Int,
    val species: String,
    val cameraName: String,
    val detectionZone: String,
    val recommendedAction: String
)

enum class SeverityLevel {
    CRITICAL, WARNING, INFO
}

enum class IncidentStatus {
    ACTIVE_BREACH, RESOLVING, RESOLVED
}

data class OrderItem(
    val id: String,
    val orderNumber: String,
    val serviceTitle: String,
    val status: OrderStatus,
    val date: String,
    val time: String,
    val address: String,
    val price: Long,
    val technicianName: String,
    val warrantyUntil: String? = null
)

enum class OrderStatus {
    PENDING, IN_PROGRESS, EN_ROUTE, COMPLETED, CANCELLED
}

data class TechnicianJob(
    val id: String,
    val code: String,
    val clientName: String,
    val address: String,
    val serviceType: String,
    val pestType: String,
    val severity: String,
    val status: JobStatus,
    val scheduleTime: String,
    val phone: String,
    val notes: String
)

enum class JobStatus {
    ASSIGNED, IN_PROGRESS, COMPLETED
}

data class ArticleItem(
    val id: String,
    val title: String,
    val snippet: String,
    val category: String,
    val readTime: String,
    val date: String
)

data class ChatMessage(
    val id: String,
    val text: String,
    val isUser: Boolean,
    val timestamp: String
)

data class CameraItem(
    val id: String,
    val name: String,
    val rtspUrl: String,
    val location: String,
    val status: String,
    val fps: Int,
    val resolution: String,
    val sensitivity: Float,
    val isAiActive: Boolean
)

data class EsgMetric(
    val label: String,
    val value: String,
    val change: String,
    val unit: String,
    val isPositive: Boolean
)

data class OnboardingSlide(
    val title: String,
    val description: String,
    val badgeText: String
)

data class PointF2D(
    val x: Float,
    val y: Float
)

File: lib/models/data_models.dart
import 'package:flutter/material.dart';

enum SeverityLevel { critical, warning, info }
enum IncidentStatus { activeBreach, resolving, resolved }
enum OrderStatus { pending, enRoute, inProgress, completed, cancelled }
enum JobStatus { assigned, inProgress, completed }

class ServiceItem {
  final String id;
  final String title;
  final String subtitle;
  final int price;
  final String description;

  const ServiceItem({
    required this.id,
    required this.title,
    required this.subtitle,
    required this.price,
    required this.description,
  });
}

class IncidentItem {
  final String id;
  final String code;
  final String title;
  final String timestamp;
  final String location;
  final SeverityLevel severity;
  IncidentStatus status;
  final int confidence;
  final String species;
  final String cameraName;
  final String detectionZone;
  String recommendedAction;

  IncidentItem({
    required this.id,
    required this.code,
    required this.title,
    required this.timestamp,
    required this.location,
    required this.severity,
    required this.status,
    required this.confidence,
    required this.species,
    required this.cameraName,
    required this.detectionZone,
    required this.recommendedAction,
  });
}

class OrderItem {
  final String id;
  final String orderNumber;
  final String serviceTitle;
  final OrderStatus status;
  final String date;
  final String time;
  final String address;
  final int price;
  final String technicianName;
  final String warrantyUntil;

  const OrderItem({
    required this.id,
    required this.orderNumber,
    required this.serviceTitle,
    required this.status,
    required this.date,
    required this.time,
    required this.address,
    required this.price,
    required this.technicianName,
    required this.warrantyUntil,
  });
}

class TechnicianJob {
  final String id;
  final String code;
  final String clientName;
  final String address;
  final String serviceType;
  final String pestType;
  final String severity;
  JobStatus status;
  final String scheduleTime;
  final String phone;
  final String notes;

  TechnicianJob({
    required this.id,
    required this.code,
    required this.clientName,
    required this.address,
    required this.serviceType,
    required this.pestType,
    required this.severity,
    required this.status,
    required this.scheduleTime,
    required this.phone,
    required this.notes,
  });
}

class ArticleItem {
  final String id;
  final String title;
  final String snippet;
  final String category;
  final String readTime;
  final String date;

  const ArticleItem({
    required this.id,
    required this.title,
    required this.snippet,
    required this.category,
    required this.readTime,
    required this.date,
  });
}

class CameraItem {
  final String id;
  final String name;
  final String rtspUrl;
  final String location;
  final String status;
  final int fps;
  final String resolution;
  double sensitivity;
  bool isAiActive;

  CameraItem({
    required this.id,
    required this.name,
    required this.rtspUrl,
    required this.location,
    required this.status,
    required this.fps,
    required this.resolution,
    required this.sensitivity,
    required this.isAiActive,
  });
}

class EsgMetric {
  final String label;
  final String value;
  final String change;
  final String unit;
  final bool isPositive;

  const EsgMetric({
    required this.label,
    required this.value,
    required this.change,
    required this.unit,
    required this.isPositive,
  });
}

class ChatMessage {
  final String id;
  final String text;
  final bool isUser;
  final String timestamp;

  const ChatMessage({
    required this.id,
    required this.text,
    required this.isUser,
    required this.timestamp,
  });
}
